react-app/src/providers/AppLocaleProvider.tsx

```diff
@@ -9,36 +9,69 @@ import {
   MessageID,
   Translations,
 } from "../i18n/LocaleModel";
+import { useLocalStorage } from "../hooks/useLocalStorage";
+
+const LOCALE_STORAGE_KEY = "LS/Locale";
 
 interface AppLocaleProviderProps {
   children?: ReactNode;
 }
 
+interface LocalizationContextValue {
+  setLocale: (locale: Locale) => void;
+}
+
+const LocalizationContext = React.createContext<LocalizationContextValue>(
+  null as any
+);
+
 const AppLocaleProvider: React.FC<AppLocaleProviderProps> = (props) => {
   const { children } = props;
 
-  // TODO: keep user local settings
-  const [locale, _setLocale] = useState<Locale>(Locale.en);
+  const [localePref, setLocalePref] =
+    useLocalStorage<Locale>(LOCALE_STORAGE_KEY);
+  const [appLocale, setAppLocale] = useState<Locale>(localePref ?? Locale.en);
```

Comment:
keeping two states looks a little weird to me 🤔

Reply:
should we handle default value fallback at the useMemo instead? (along with checking if the local is valid, see below)